### DISCUSSION TITLE
Setting up Jellyfin for step-debugging on Windows, with plugin

### DISCUSSION THREAD

**Discussion by voltagex** [jellyfin/jellyfin#11903] [Q&A]
I am trying to debug an issue (https://github.com/jellyfin/jellyfin-plugin-dlna/issues/56)

It would be very useful to be able to sit in Visual Studio and hit breakpoints.

So far I have worked out:

* Clone and build Jellyfin in VS2022, select the nowebclient option
* Clone and build jellyfin-web (after installing nodejs via winget install OpenJS.NodeJS.LTS
* Go to localhost:8080, "log in" to a server at localhost:8096

---
* Get stuck, because there's no configured user by default and I'm not being sent to the first-start wizard
* Only after *copying a database entry from a configured server and removing an allowed device check did I log in and see the wizard*
* After going through the wizard and changing my password, I am no longer an admin
* Copy from the Permissions table on the working server, restart and I'm an admin now
---
* Add a `Debugger.Break()` line to the plugin because I see no other way of debugging it
* Manually build the DLNA plugin, put it in the right path
* Restart, hit breakpoint after ~1 hour of setup

This is not actually the most difficult an open source project has been to build, run and debug.

I am happy to contribute to Jellyfin, but there has to be an easier way to get up and running.

**Comment by anthonylavado** [comment]
When you "got stuck", was there anything in particular on the screen? Was it a server select screen perhaps?

**Reply by voltagex** [reply]
@anthonylavado I really should have kept better notes - I will retry this week. I don't think it was the server select screen, I selected the localhost:8096 server on that screen and was then prompted for a username and password for a user that didn't exist. Perhaps because I was running the UI and server on different ports?

**Reply by TheMelmacian** [reply]
If you run the server and web-gui separately you have to use different ports. If one service already uses the port, the other one would abort the startup process with the error message "port is already in use" or something similar.
So this is perfectly normal and not the reason for your problem.

**Comment by mjohl** [comment]
@voltagex I downloaded the repo, hit build and ran the nowebclient profile. Looking at the code, I used "http://localhost:8096/Startup/User" to get the user. Then used the windows client to launch and entered the username, no password. And I was in. I already have a instance running, so I just had to change all references to 8096 to 8095 for the server port and port 7359 I changed to 7360 in the AutoDiscoveryHost.

The ports in my opinion should be launch parameter for debug, but that I still need to look into.

![Screenshot 2024-07-07 024639](https://github.com/jellyfin/jellyfin/assets/4193377/1cef5cf7-5b50-4760-a22b-926889daaf9d)
